paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/CustomerSheetTest.kt

```diff
@@ -143,6 +143,64 @@ internal class CustomerSheetTest {
         page.clickConfirmButton()
     }
 
+    @Test
+    fun testDeleteSavedCard() = activityScenarioRule.runCustomerSheetTest(
+        integrationType = integrationType,
+        customerSheetTestType = CustomerSheetTestType.AttachToSetupIntent,
+        resultCallback = { result ->
+            assertThat(result).isInstanceOf(CustomerSheetResult.Selected::class.java)
+
+            val selected = result as CustomerSheetResult.Selected
+
+            assertThat(selected.selection).isInstanceOf(PaymentOptionSelection.PaymentMethod::class.java)
+
+            val paymentMethodSelection = selected.selection as PaymentOptionSelection.PaymentMethod
+
+            val card = paymentMethodSelection.paymentMethod.card
+
+            assertThat(card?.last4).isEqualTo("4242")
+            assertThat(card?.brand).isEqualTo(CardBrand.Visa)
+        }
+    ) { context ->
+        context.scenario.onActivity {
+            PrefsTestStore(it).clear()
+        }
+
+        networkRule.enqueue(
+            retrieveElementsSessionRequest(),
+        ) { response ->
+            response.testBodyFromFile("elements-sessions-requires_payment_method.json")
+        }
+
+        networkRule.enqueue(
+            retrievePaymentMethodsRequest(),
+            cardPaymentMethodsParams(),
+        ) { response ->
+            response.testBodyFromFile("payment-methods-get-success.json")
+        }
+
+        networkRule.enqueue(
+            retrievePaymentMethodsRequest(),
+            usBankAccountPaymentMethodsParams(),
+        ) { response ->
+            response.testBodyFromFile("payment-methods-get-success-empty.json")
+        }
+
+        context.presentCustomerSheet()
+
+        networkRule.enqueue(
+            detachRequest(),
+        ) { response ->
+            response.testBodyFromFile("payment-methods-detach.json")
+        }
+
+        page.clickEditButton()
+        page.clickDeleteButton(forEndsWith = "4242")
+        page.clickDialogRemoveButton()
+
+        context.markTestSucceeded()
```

Comment:
should we test that the saved card is removed from the UI or anything here too?

Reply:
I actually had some testing logic for this but seemingly ran into Compose testing issues where the element is visibly removed from the screen on the emulator but `ComposeTestRule` is still somehow able to find the component. I'm planning on evaluating what is going on there outside of this PR.

Reply:
oh interesting. Figuring it out outside of this PR sounds good!